lib/hexapdf/type/image.rb

```diff
@@ -254,6 +257,48 @@ def png_chunk(type, data = '')
         [data.length].pack("N") << type << data << [Zlib.crc32(data, Zlib.crc32(type))].pack("N")
       end
 
+      # Writes the image as TIFF to the given IO stream.
+      def write_tiff(io, info)
+        stream = stream_source
+
+        # header
+        io << ['49492A00'].pack('H*')
+
+        # offset to IFD
+        pad_size = stream.length % 2
+        io << [stream.length + 8 + pad_size].pack('L')
+
+        # data
+        io << Filter.string_from_source(stream)
+
+        # pad to word boundary
+        io << [0].pack('C') * pad_size
+
+        decode_params = [self[:DecodeParms]].flatten[0]
+        compression = if decode_params[:K] > 0 # G3-2D
+                        3
+                      elsif decode_params[:K] < 0 # G4-2D
+                        4
+                      else # G3-1D
+                        2
+                      end
+
+        # IFD entries
+        ifd_entries = [
+            [0x0100, 3, 1, info.width], # width
+            [0x0101, 3, 1, info.height], # height
+            [0x0102, 3, 1, info.bits_per_component], # bits per sample
```

Comment:
Do we need the entry since `info.bits_per_component` should always be 1 for CCITTFaxDecode? Also, according to the spec this is only used with `compression == 1` and compression never has that value.

Reply:
If `0x0102` is not present, macOS Preview app won't open the file...

Reply:
Since it seems that this won't result in an invalid TIFF file, I'm okay with including this information. But add a comment why it is included.